### ISSUE TITLE
[HELP] Enabling BT_RFCOMM, HID, Injection, Nexmon features in menuconfig 

### ISSUE THREAD

**Issue by Mr-x2020** [lavanoid/Hammerhead_Kali#1]
This might be super random but I need your help please, There are features are not working in Nethunter and I have to enable them in the Android kernel source, I started to compile the kernel source of my device and I tried to enable BT_RFCOMM, HID, Injection, Nexmon features in menuconfig but I failed because I don't know what these features named in menuconfig there are too many options. I need help finding the path to BT_RFCOMM, HID, Injection, Nexmon features in menuconfig

**Comment by lavanoid** [comment]
` HID, Injection, Nexmon ` These are patches which are not in the menuconfig. There is a dedicated (albeit outdated) tutorial for patching the kernel here: https://github.com/offensive-security/kali-nethunter/wiki/Modifying-the-Kernel

This repo contains the patches for HID and injection (hid.patch, and the injection patch is mentioned in the `build.sh`, here: http://patches.aircrack-ng.org/mac80211.compat08082009.wl_frag+ack_v1.patch ). However I am unable to compile the kernel (hence why this is a WIP) due to multiple errors being thrown out. I cannot compile the kernel even without modifications, so I need to do some digging to find out why. Maybe it's toolchain related, I'm not entirely sure.

I hope this helps.

**Comment by lavanoid** [comment]
Also, Nexmon can be installed via Hijacker (though it doesn't install all of the required utilities, just enough to get monitor mode/injection working): https://store.nethunter.com/en/packages/com.hijacker/

The Nexmon app included in the Nethunter store, doesn't have any of the modified firmware files or utilities included for whatever reason. Might be worth trying this build of Nexmon: https://github.com/seemoo-lab/nexmon/issues/477#issuecomment-829212527

**Comment by Mr-x2020** [comment]
thank you for your quick reply, I am also have Google nexus 5 and I've been trying to enabling some features and compiling the follwing source code: https://github.com/LineageOS/android_kernel_lge_hammerhead/tree/cm-14.1

what about rfcomm module?

**Comment by lavanoid** [comment]
I don't believe I've enabled the rfcomm module before, but I can take a look once I have some spare time if you want. I'll try compiling the kernel you linked, with the required patches/drivers enabled and report my findings.

**Comment by lavanoid** [comment]
Is there a particular reason why you're using that version of Android by the way? I've been trying to port Binkybears patches over to Android 10 (Y-Cable charging, HID, disk/mass storage emulation etc).

**Comment by Mr-x2020** [comment]
thank you for your help , I have teamviewer setted up if you wanna check it out tomorrow in my workplace  :) ,
I use nougat version because there is already builded image of nethunter for my nexus 5 
![image](https://user-images.githubusercontent.com/65666323/122681456-ee41c080-d1e3-11eb-992c-65c361a5c034.png)

**Comment by lavanoid** [comment]
That makes sense. If you have a Discord, we can discuss things further there.

**Comment by Mr-x2020** [comment]
here is my discord username elkhazrajy#3454

**Comment by lavanoid** [comment]
FR sent
